src/libsystemd/sd-event/event-util.c

```diff
@@ -91,3 +91,16 @@ int event_source_disable(sd_event_source *s) {
 
         return sd_event_source_set_enabled(s, SD_EVENT_OFF);
 }
+
+int event_source_is_enabled(sd_event_source *s) {
+        int enabled, r;
+
+        if (!s)
+                return false;
+
+        r = sd_event_source_get_enabled(s, &enabled);
+        if (r < 0)
+                return r;
+
+        return enabled != SD_EVENT_OFF;
```

Comment:
TODO: we could also change `sd_event_source_get_enabled` to return `1` in this case. This would be generally useful. But this PR is long already, so let's leave that for later.

Reply:
https://github.com/systemd/systemd/pull/10791